src/goto-instrument/contracts/contracts.cpp

```diff
@@ -252,13 +252,30 @@ void code_contractst::check_apply_loop_contracts(
     loop_head,
     add_pragma_disable_assigns_check(generated_code));
 
-  // havoc the variables that may be modified
-  assignst assigns;
+  assignst to_havoc;
+
   if(assigns_clause.is_nil())
   {
+    // No assigns clause was specified for this loop.
+    // Infer memory locations assigned by the loop from the loop instructions
+    // and the inferred alisaing relation.
     try
     {
-      get_assigns(local_may_alias, loop, assigns);
+      get_assigns(local_may_alias, loop, to_havoc);
+      log.warning() << "No loop assigns clause provided. Inferred targets {";
+      // Add inferred targets to the loop assigns clause.
+      bool ran_once = false;
+      for(auto &target : to_havoc)
+      {
+        if(ran_once)
+          log.warning() << ", ";
+        ran_once = true;
+        log.warning() << from_expr(ns, target.id(), target);
+        loop_assigns.add_to_write_set(target);
+      }
+      log.warning()
+        << "}. Please specify an assigns clause if verification fails."
+        << messaget::eom;
```

Comment:
Is there any chance of the assertion being generated to include that information? So that we end up with something a long the lines of "Auto-generated assigns set is complete: FAILURE"